apps/hash-graph/lib/graph/src/api/rest/json_schemas/entity_type.json

```diff
@@ -10,63 +10,150 @@
+      }
+    },
     "examples": {
-      "$comment": "Example Entity instances",
       "type": "array",
       "items": {
-        "type": "object",
-        "propertyNames": {
-          "$comment": "Property names must be a valid URL to a Property Type",
-          "type": "string",
-          "format": "uri"
-        }
+        "type": "object"
       }
     },
-    "properties": { "$ref": "#/$defs/propertyTypeObject" },
+    "properties": {
+      "$ref": "#/definitions/PropertyTypeObject"
+    },
     "required": {
       "type": "array",
       "items": {
-        "type": "string",
-        "format": "uri"
+        "$ref": "#/definitions/BaseUrl"
       }
     },
-    "links": { "$ref": "#/$defs/linkTypeObject" }
+    "links": {
+      "type": "object",
+      "propertyNames": {
+        "$ref": "#/definitions/VersionedUrl"
+      },
+      "patternProperties": {
+        "^.*$": {
+          "type": "object",
+          "properties": {
+            "type": {
+              "type": "string",
+              "enum": ["array"]
+            },
+            "ordered": {
+              "type": "boolean",
+              "default": false
+            },
+            "items": {
+              "$ref": "#/definitions/OneOfEntityTypeReference"
+            },
+            "minItems": {
+              "type": "integer",
+              "minimum": 0
+            },
+            "maxItems": {
+              "type": "integer",
+              "minimum": 0
+            }
+          },
+          "required": ["type", "ordered", "items"],
+          "additionalProperties": false
+        }
+      }
+    }
   },
   "additionalProperties": false,
-  "required": ["$schema", "kind", "$id", "type", "title", "properties"],
-  "$defs": {
-    "propertyTypeObject": {
+  "required": ["$schema", "kind", "type", "$id", "title", "properties"],
+  "definitions": {
+    "VersionedUrl": {
+      "title": "Versioned URL",
+      "description": "The versioned URL of a Block Protocol ontology type (the $id of the schema). It should be of the form `${baseUrl}v/${versionNumber}`",
+      "type": "string",
+      "format": "uri",
+      "maxLength": 2048
+    },
+    "BaseUrl": {
+      "title": "Base URL",
+      "description": "The base URL of a Block Protocol ontology type (the $id of the schema, without the versioned suffix). It should a valid URL, with a trailing slash.",
+      "type": "string",
+      "format": "uri",
+      "maxLength": 2048
+    },
+    "EntityTypeReference": {
+      "type": "object",
+      "properties": {
+        "$ref": {
+          "$ref": "#/definitions/VersionedUrl"
+        }
+      },
+      "required": ["$ref"],
+      "additionalProperties": false
+    },
+    "PropertyTypeReference": {
+      "type": "object",
+      "properties": {
+        "$ref": {
+          "$ref": "#/definitions/BaseUrl"
+        }
+      },
+      "required": ["$ref"],
+      "additionalProperties": false
+    },
+    "OneOfEntityTypeReference": {
```

Comment:
Splitting this out does not help readability. I'd prefer to define it inline.

Reply:
Sure, essentially just copied it from the original.

Reply:
fixed in [9ccbd86](https://github.com/hashintel/hash/pull/2838/commits/9ccbd8673e92eede8b25ed19917a6f23ec116aa0)